src/state/slices/opportunitiesSlice/types.ts

```diff
@@ -78,14 +101,18 @@ export type UserStakingOpportunity =
 export type UserStakingOpportunityWithMetadata = UserStakingOpportunity & OpportunityMetadata
 
 // The AccountId of the staking contract in the form of chainId:accountAddress
-export type StakingId = Nominal<string, 'StakingId'> & AssetId
+export type StakingIdBase = Nominal<string, 'StakingId'> & AssetId
+// Sometimes, an AssetIdish (i.e a a CAIP-19) is not enough to uniquely identify a staking opportunity
+// So we add a StakingKey to the StakingId to make it unique
+export type StakingKey = Nominal<string, 'StakingKey'> & string
```

Comment:
```suggestion
export type StakingKey = Nominal<string, 'StakingKey'>
```

previous `& AssetId` pattern was just such that `StakingId` becomes a superset of `AssetId`

Reply:
e05b6759b